### DISCUSSION TITLE
Improve the user experience

### DISCUSSION THREAD

**Discussion by jeaunrg** [Hboni/francis#37] [Ideas]
UX remains very important to my mind, so:
- we should add icons to replace some button to make the interface more user-friendly. 
- some gif for loading if processes are made in the background.
- and better colors, those I choose are lousy 🤢
Maybe one day...

**Comment by Hboni** [comment]
I agree with your thought, I also feel not really happy with the actual interface. 
We must discuss here all the improvements that we want for the next releases. Then we can decide which ones are more interesting and add them to the future release, or at least choose a certain amount of upgrades for the next version (1.1?).

To make a complete list of what can be improved IMO:
1. image/gif/color change to display when a node is running or not
2. image/color change to say if node updated (in case of parent update)
3. change selection visualization, it does not POP to the eye what is selected
4. create a logo?
5. change colors

Other ideas? I think that 1 and 3 could be good for the next release (not the one we are finishing), what do you think?

**Reply by jeaunrg** [reply]
yep I agree all along the line. You should have a quick look at my little demo app https://github.com/jeaunrg/ModelViewPresenter_app for the gif we could use the same widget

**Reply by Hboni** [reply]
I saw your demo app, it looks great and could be good on Francis. But to organize tasks for next releases, we must agree upon features for next release.
The idea is to discuss about next features, and only select which one we wants to include in next release. I start another discussion to talk about features only, to keep interface improvement here.
6. I was wondering if it is also possible to change a little bit junctions to be more convenient in case of nodes not strictly on the right of parent?

**Reply by jeaunrg** [reply]
Everything is possible, what do you have in mind more specifically?

**Reply by Hboni** [reply]
I have two ideas for junctions :
- Each junction can move around the node, and its direction aligned with both node's centers. But in this case, we would have a linear junction more or less
- Keep what exists but where the junction pops out/in a node, begin a little bit straight before changing direction, to be able to see it more clearly. An example below with what I think on the right:
![node_junction](https://user-images.githubusercontent.com/27290853/115290234-9b9c4780-a153-11eb-8dfa-705ed48e5e1b.png)

**Reply by jeaunrg** [reply]
I prefer the first idea considering that the position of two nodes could be from right to left, but in this case the direction of the arrow will be hidden. Maybe we could change the arrow texture like that ?  (ref #43) 
![rect923](https://user-images.githubusercontent.com/57048964/115439525-f6df4000-a20e-11eb-9030-58f52650c0c4.png)

**Comment by Hboni** [comment]
As you wish, both ideas are good to me, they can increase readability. Do not hesitate to put a note/issue in the Francis_1.1 project if you plan to work on it for the next release ;)

**Reply by jeaunrg** [reply]
ok!
